harness/determined/common/experimental/project.py

```diff
@@ -0,0 +1,214 @@
+        resp = bindings.get_GetProject(session=self._session, id=self.id)
+        project_bindings = resp.project
+
+        self._hydrate(project_bindings)
+
+    def list_experiments(self) -> List["experiment.Experiment"]:
+        def get_with_offset(offset: int) -> bindings.v1GetExperimentsResponse:
+            return bindings.get_GetExperiments(
+                session=self._session, projectId=self.id, offset=offset
+            )
+
+        exp_bindings = itertools.chain.from_iterable(
+            r.experiments for r in api.read_paginated(get_with_offset)
+        )
+        return [experiment.Experiment._from_bindings(exp, self._session) for exp in exp_bindings]
+
+    def set_description(self, description: str) -> None:
+        """Set the description of the project.
+
+        The attribute will be changed both on master and this local object.
+
+        Args:
+            description: The description to set.
+        """
+        patch_body = bindings.v1PatchProject(description=description)
+        resp = bindings.patch_PatchProject(session=self._session, id=self.id, body=patch_body)
+
+        self.description = resp.project.description
+
+    def set_name(self, name: str) -> None:
+        """Set the name of the project.
+
+        The attribute will be changed both on master and this local object.
+
+        Args:
+            name: The name to set.
+        """
+        patch_body = bindings.v1PatchProject(name=name)
+        resp = bindings.patch_PatchProject(session=self._session, id=self.id, body=patch_body)
+
+        self.name = resp.project.name
+
+    def archive(self) -> None:
+        """Set the project to archived (archived = True).
+
+        As with other setters, this will change the attribute both on master and this local object.
+        """
+        bindings.post_ArchiveProject(session=self._session, id=self.id)
+        self.archived = True
+
+    def unarchive(self) -> None:
+        """Set the project to unarchived (archived = False).
+
+        As with other setters, this will change the attribute both on master and this local object.
+        """
+        bindings.post_UnarchiveProject(session=self._session, id=self.id)
+        self.archived = False
+
+    def add_note(self, name: str, contents: str) -> None:
+        """Add a note to the project.
+
+        Because there is not yet functionality on the backend to add a single note, this method:
+        1. fetches current notes for this project from the master.
+        2. adds the new note to the list of notes.
+        3. sends the updated list of notes to the master.
+
+        WARNING:
+        When it exits, the notes attribute of this object will be the same notes updated on master.
+        This set of notes might reflect changes to the project that have happened since the project
+        was last hydrated from master.
+
+        Args:
+            name: The name of the note.
+            contents: The contents of the note.
+        """
```

Comment:
"""Add a note to the project.
        Because there is no functionality on the backend to add a single note, this method fetches current notes, adds a new note, and updates the master.
        Post-execution, the object's notes attribute matches the updated master's notes, 
        possibly reflecting changes to the project that have happened since the project was last hydrated from master.
        Args:
            name: Note's title.
            contents: Note's content.
        """

Reply:
I like the warning text you've suggested and have updated the docstring, but the I think the textual description of what the method does is a little less clear without the enumeration. What are your thoughts here?

Reply:
I was going for brevity. The enumeration could help. I like to save the imperative mood for step-by-step instructions. I was thinking if the docstring gets rendered in a page of content on the docs site via Sphinx autodocs, they may, at first glance, think these are instructions. But the enumeration (or bullet points) you had is definitely clear in the context of just looking at the script.

Reply:
I see that the google style guide also talks about imperative mood. To clarify, I use numbered steps when providing step by step instructions written in the imperative mood.

Reply:
looks good, i'll set this to resolved